learntools/machine_learning/nbs/ch3.ipynb

```diff
@@ -0,0 +1,292 @@
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# The lines below will show you a hint or the solution.\n",
+    "# step_1.hint() \n",
+    "# step_1.solution()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 2: Create DataFrame of Predictors\n",
+    "Now you will create a DataFrame called `X` holding the predictors' data.\n",
+    "\n",
+    "Since you want only some columns from the original data, you'll first create a list with the names of the columns you want in `X`.\n",
+    "\n",
+    "You'll use just the following columns in the list (you can copy and paste the whole list to save some typing, though you'll still need to add quotes):\n",
+    "    * LotArea\n",
+    "    * YearBuilt\n",
+    "    * 1stFlrSF\n",
+    "    * 2ndFlrSF\n",
+    "    * FullBath\n",
+    "    * BedroomAbvGr\n",
+    "    * TotRmsAbvGrd\n",
+    "\n",
+    "After you've created that list of predictors, use it to create the DataFrame that you'll use when fitting the model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# Create the list of predictors below\n",
+    "# predictor_names = ___\n",
+    "\n",
+    "# select data corresponding to predictors in predictor_names\n",
+    "#X = _\n",
+    "\n",
+    "step_2.check()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# step_2.hint()\n",
+    "# step_2.solution()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Review Data\n",
+    "Before building a model, take a quick look at your predictors to see if they seem sensible"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# Review data\n",
+    "# print description or statistics from X\n",
+    "#print(_)\n",
+    "\n",
+    "# print the top few lines\n",
+    "#print(_)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 3: Specify and Fit Model\n",
+    "Create a `DecisionTreeRegressor` and save it iowa_model. Ensure you've done the relevant import from sklearn to run this command.\n",
+    "\n",
+    "Then fit the model you just created using the data in `X` and `y` that you saved above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# from _ import _\n",
+    "#specify the model. \n",
+    "#For model reproducibility, set a numeric value for random_state when specifying the model\n",
+    "iowa_model = _\n",
+    "\n",
+    "# Fit the model\n",
+    "_\n",
+    "\n",
+    "step_3.check()"
+   ]
```

Comment:
I tried tackling this step incrementally, first trying to fix the import then running the code cell to see if it worked. The Incorrect message I got was really weird. It said something like `Incorrect: Expected type object of type DecisionTreeRegressor but got an object of type Pandas.IndexedSeries`. When I ran it again, it said the type was `str`. It took me a step to wrap my head around this - what's going on is that the Python interpreter stores the value of the last expression in the variable `_`, so while it seems like `_` is just a blank line saying "your answer goes here", it actually has some very wacky implications. 

I don't know if it's worth changing at this point, but it's at least something to be aware of in the future (and in case it comes up in any forum posts or comments).